docs/overview.rst

```diff
@@ -0,0 +1,83 @@
+.. _concepts:
+
+Overview
+========
+The following is a walk-through of the important concepts in the access control
+system.
+
+Users & roles
+-------------
+First we have **subjects** which can be granted access to a protected resource.
+
+- **User**: Represents an authenticated account in the system.
+- **Role**: Represents a job function. Roles are created by e.g. system
+  administrators and defined by a name. Users can be assigned zero or more
+  roles.
+- **System role**: Represents special roles that are created and defined by the
+  system and automatically assigned to users (i.e. system roles cannot be
+  created and defined by system administrators).
+
+Permissions and needs
+---------------------
+Second, we have two entities to describe access control:
+
+- **Need**: A need represents the smallest level of access control. It is very
+  generic and can express statements such as *"has admin role"* and
+  *"has read access to record 42"*.
+- **Permission**: Represents a set of required *needs*, any of which should
+  be fulfilled to access a resouce. E.g. a permission can combine the two
+  statements above into **"has admin role or has read access to record 42"**.
+
+The concept of a *need* can be somewhat hard to grasp at first, so let's
+dive in to describe how a need is used. Essentially *needs* are used to express
+a) what a permission require and b) what a user provides, i.e.:
+
+- A permission **requires** a set of needs.
+- A user **provides** a set of needs.
+
+Thus, checking if a user can access a resource protected by a permission
+amounts to checking for a non-empty **intersection** between the above sets.
```

Comment:
Since the other parts are talking about sets, I stayed in the domain and used intersection. Probably a diagram like https://en.wikipedia.org/wiki/Intersection_(set_theory)#/media/File:Venn0001.svg with some descriptive text may ease it? Let me see if I can come up with a better explanation.